hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieNativeAvroHFileReader.java

```diff
@@ -421,17 +436,19 @@ private static class RecordByKeyIterator implements ClosableIterator<IndexedReco
 
     private final Schema readerSchema;
     private final Schema writerSchema;
+    private final Option<BloomFilter> bloomFilterOption;
 
     private IndexedRecord next = null;
 
     RecordByKeyIterator(HFileReader reader, List<String> sortedKeys, Schema writerSchema,
-                        Schema readerSchema) throws IOException {
+                        Schema readerSchema, Option<BloomFilter> bloomFilterOption) throws IOException {
```

Comment:
do you think we can change this to generic Predicate argument

Reply:
This is a bloom filter object built from the existing keys in the file, so it's not a predicate itself.  The predicate is represented by the `sortedKeys`.